### PR TITLE
Feat/manage voting power

### PR DESCRIPTION
**Pull request by roshan-bisht-lab49**
Screenshot 2023-04-24 at 6 29 01 PM
Screenshot 2023-04-24 at 6 29 14 PM
Screenshot 2023-04-24 at 6 28 51 PM
Screenshot 2023-04-24 at 6 29 53 PM

### CODE REVIEW

File: src/dex-ui-components/base/Labels/MetricLabel.tsx [hunk 1]
```diff
@@ -1,21 +1,66 @@
 import { Flex, Text, Skeleton } from "@chakra-ui/react";
+import { Color } from "../../themes";
 
 interface MetricLabelProps {
   label: string;
   value: string;
   isLoading?: boolean;
+  labelLeftIcon?: React.ReactElement;
```
Comment (roshan-bisht-lab49) [inline src/dex-ui-components/base/Labels/MetricLabel.tsx:8]: We have one challenge in `metric label` the every label in `Manage Voting Power` has different `FontStyles` and `FontColor` so for now I have added optional props for that change. let me know if this works.

File: src/dex-ui/pages/Governance/ManageVotingPower/ManageGovToken.tsx [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+  });
+  manageGovTokenForm.watch(["lockAmount", "unLockAmount"]);
+
+  function onSubmit(data: InputTokenAmountData) {
+    console.log("Submit Pressed", data);
+  }
+
+  return (
+    <form>
+      <AlertDialog
```
Comment (roshan-bisht-lab49) [inline src/dex-ui/pages/Governance/ManageVotingPower/ManageGovToken.tsx:162]: Here, `Loading Dialog`, `Success Dialog` and `Error Dialog` will also come that change will come once we get the SC for this.

File: src/dex-ui/pages/Governance/ManageVotingPower/ManageVotingPower.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+import { VotingPowerCard } from "./VotingPowerCard";
+import { ManageGovToken } from "./ManageGovToken";
+import { useNavigate } from "react-router-dom";
+import { useManageVotingPower } from "./useManageVotingPower";
+import { MetricLabel } from "../../../../dex-ui-components";
+
+export const ManageVotingPower = () => {
+  const { userGOVTokenWalletBalance, isLoading, doesUserHaveGOVTokensToLock } = useManageVotingPower();
+
+  const navigate = useNavigate();
```
Comment (roshan-bisht-lab49) [inline src/dex-ui/pages/Governance/ManageVotingPower/ManageVotingPower.tsx:13]: Need to check with team what UI will come if the user wallet is not connected.

**Review by roshan-bisht-lab49 — COMMENTED** [review]
Added Helpful comments

File: src/dex-ui-components/base/Labels/MetricLabel.tsx (hunk 1 above)
Comment (damiano-melcarne-lab49) [reply src/dex-ui-components/base/Labels/MetricLabel.tsx:8]: This is ok for now. I believe the metric label styles should match across the application in the final product but reach out to the design team to confirm.

File: src/dex-ui-components/base/Labels/MetricLabel.tsx [hunk 4]
```diff
@@ -1,21 +1,66 @@
...
 
   return (
     <Flex flexDirection="column" gap="1">
-      <Text textStyle="h4" opacity="0.8">
-        {label}
-      </Text>
+      <Flex flexDirection="row" alignItems="center" gap="1">
+        {props.labelLeftIcon ? props.labelLeftIcon : null}
+        <Text
+          textStyle={props.labelTextStyle ? props.labelTextStyle : "h4"}
```
Comment (damiano-melcarne-lab49) [inline src/dex-ui-components/base/Labels/MetricLabel.tsx:29]: You can use prop destructing with default values to reduce the need for ternary operators:

`const { label, value, isLoading = false, labelTextStyle = "h4", labelTextColor = Color.Neutral._900, etc... } = props;`

It's a personal preference but I think the destructing approach is cleaner.

File: src/dex-ui/pages/Governance/ManageVotingPower/ManageGovToken.tsx [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+                ),
+                placeholder: "",
+                register: {
+                  ...props.form.register("lockAmount", {
+                    required: { value: true, message: "Amount cannot be more than available balance." },
+                  }),
+                },
+              }}
+              isInvalid={Boolean(props.form.formState.errors.lockAmount)}
+              errorMessage={props.errors.lockAmount && props.errors.lockAmount.message}
```
Comment (damiano-melcarne-lab49) [inline src/dex-ui/pages/Governance/ManageVotingPower/ManageGovToken.tsx:102]: Can use optional chaining here: props.errors.lockAmount?.message

**Review by damiano-melcarne-lab49 — CHANGES_REQUESTED** [review]

File: src/dex-ui-components/base/Labels/MetricLabel.tsx (hunk 4 above)
Comment (roshan-bisht-lab49) [reply src/dex-ui-components/base/Labels/MetricLabel.tsx]: Thanks for the helpful comment, I have addressed this.

File: src/dex-ui/pages/Governance/ManageVotingPower/useManageVotingPower.tsx [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+  function getErrorDialogMessage(): string {
+    return "";
+  }
+  const errorDialogMessage = getErrorDialogMessage();
+
+  function getSuccessMessage(): string {
+    return "";
+  }
+  const successMessage = getSuccessMessage();
+  // function getHashScanLink(): string | undefined {
```
Comment (damiano-melcarne-lab49) [inline src/dex-ui/pages/Governance/ManageVotingPower/useManageVotingPower.tsx:32]: Remove the commented code if it is no longer needed.

**Review by damiano-melcarne-lab49 — CHANGES_REQUESTED** [review]

File: src/dex-ui/pages/Governance/ManageVotingPower/useManageVotingPower.tsx (hunk 6 above)
Comment (roshan-bisht-lab49) [reply src/dex-ui/pages/Governance/ManageVotingPower/useManageVotingPower.tsx]: Done

**Review by damiano-melcarne-lab49 — APPROVED** [review]

**Pull request merged**